### PR TITLE
Product view controller UI collection view cells

### PR DESCRIPTION
**Pull request by RSKuse**

### CODE REVIEW

File: Starbucks/Controllers/ProductsViewController/ProductsViewController.swift [hunk 1]
```diff
@@ -8,10 +8,78 @@
...
+        cell.productNameLabel.text = product.name
+        return cell
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let numberOfItemsPerRow: CGFloat = 2
+        let spacingBetweenItems: CGFloat = 8
+        let totalSpacing = (2 * spacingBetweenItems) + ((numberOfItemsPerRow - 1) * spacingBetweenItems)
+        let width = (collectionView.bounds.width - totalSpacing) / numberOfItemsPerRow
+        return CGSize(width: 171, height: 230)
```
Comment (mhlangagc) [inline Starbucks/Controllers/ProductsViewController/ProductsViewController.swift]: @RSKuse I noticed you calculated the width and height but did not use it. You hard coded the CGSize.

File: Starbucks/Controllers/ProductsViewController/ProductsViewController.swift [hunk 2]
```diff
@@ -8,10 +8,78 @@
...
+        view.backgroundColor = UIColor.white
+        setupConstraints()
+        registerCollectionCells()
+    }
+    
+    func setupConstraints() {
+        view.addSubview(productCollectionView)
+        
+        productCollectionView.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 20).isActive = true
+        productCollectionView.rightAnchor.constraint(equalTo: view.rightAnchor, constant: -20).isActive = true
```
Comment (mhlangagc) [inline Starbucks/Controllers/ProductsViewController/ProductsViewController.swift:48]: You could alternatively use the `contentInset` to define these edges. Constraints are also fine.

File: Starbucks/Controllers/HomeViewController/Components/TableSectionHeaderView.swift [hunk 3]
```diff
@@ -10,6 +10,12 @@ import UIKit
 
 class TableSectionHeaderView: UIView {
     
+    var title: String? {
+        didSet {
+            titleLabel.text = title
+        }
+    }
```
Comment (mhlangagc) [inline Starbucks/Controllers/HomeViewController/Components/TableSectionHeaderView.swift:17]: Perfect. 💯

File: Starbucks/Controllers/HomeViewController/HomeViewController+TableView.swift [hunk 4]
```diff
@@ -104,7 +104,7 @@ extension HomeViewController {
         
         // Comment: Adds the tappability of the seeAllButton
         headerView.seeAllButton.addTarget(self, action: #selector(seeAllButtonTapped), for: .touchUpInside)
-        headerView.titleLabel.text = starBucksDatabase.categoriesArray[section]
+        headerView.title = starBucksDatabase.categoriesArray[section]
```
Comment (mhlangagc) [inline Starbucks/Controllers/HomeViewController/HomeViewController+TableView.swift:110]: 🥇 Nice Update.

File: Starbucks/Controllers/ProductsViewController/ProductCollectionViewCell.swift [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+import Foundation
+import UIKit
+
+class ProductCollectionViewCell: UICollectionViewCell {
+    
+    lazy var productImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.contentMode = .scaleAspectFill
+        imageView.image = UIImage(named: "latte")
```
Comment (mhlangagc) [inline Starbucks/Controllers/ProductsViewController/ProductCollectionViewCell.swift:16]: You can add this image to the `StarbucksImages` class

**Review by mhlangagc — CHANGES_REQUESTED** [review]

File: Starbucks/Controllers/ProductsViewController/ProductsViewController.swift (hunk 2 above)
Comment (RSKuse) [reply Starbucks/Controllers/ProductsViewController/ProductsViewController.swift:48]: for this one I sort of forgot how to implement the ContentInset.. but I will try

File: Starbucks/Controllers/ProductsViewController/ProductsViewController.swift (hunk 1 above)
Comment (mhlangagc) [reply Starbucks/Controllers/ProductsViewController/ProductsViewController.swift]: The result of the hard coding when I run the build on a iPhoneSE. @RSKuse 

Simulator Screenshot - iPhone SE (3rd generation) - 2024-05-06 at 17 35 44

Comment (RSKuse) [reply Starbucks/Controllers/ProductsViewController/ProductsViewController.swift]: I do not understand here? what you  mean

**Pull request merged**